pam_duo/pam_duo.c

```diff
@@ -233,8 +233,17 @@ pam_sm_authenticate(pam_handle_t *pamh, int pam_flags,
     pam_err = PAM_SERVICE_ERR;
 
     for (i = 0; i < cfg.prompts; i++) {
-        code = duo_login(duo, user, host, flags,
-                    cfg.pushinfo ? cmd : NULL);
+        if (!cfg.domain || !cfg.domain[0]) {
+            code = duo_login(duo, user, host, flags,
+                cfg.pushinfo ? cmd : NULL);
+        } else {
+            char *duo_auth_user=(char*)malloc(strlen(user)+strlen(cfg.domain)+2);
```

Comment:
duo_auth_user is never free'd, and the malloc() return ought to be checked before use.